Spike snapshots and prototype (#59)
HANICA-DWA/sep2021-project-opossum: documentatie/Spikes/Snaphots/readme.md

# Snapshot Spike

In this we research the best way to save a page. We need this so we can annotate dynamic every-changing websites.

> ## Mhtml

Reviewer: ```suggestion
## MHTML
```

The browser API for extension (Chrome and Firefox) has the ability to save pages in mhtml.

> We haven't chosen for this due to the fact that mhtml has some difficulty with being displayed in html iframes. Doesn't work with JavaScript and images aren't stored.

Reviewer: ```suggestion
We haven't chosen for this due to the fact that iframes have difficulty displaying MHTML files. It also doesn't work with JavaScript and images and other media aren't stored.
```

In tests the page didn't come out the same as the original webpage.

## single-file html

> When researching this problem we have search for existing plugins that try to save a web pages. We've found multiple but settled with single-file because of the following.

Reviewer 2: ```suggestion
When researching this problem we have searched for existing plugins that try to save a web page. We've found multiple but settled with single-file because of the following.
```

* Custom options when saving (with or without JavaScript, compression, etc.)
* Great support
  * The GitHub repository of single-file has:
    * Wiki pages for integrating single-file API in extensions
    * Active developer that responds to user created issues and emails
    * Users can create issues
* Elaborate extension with page editor.

As the name suggests single-file plugin can save a page into a single html file. It encodes and inlines images, CSS, iframes, fonts and JavaScript. So you have everything in a single file without needing to do web requests.

> The exception is video's and audio files. It has the option to remove the link to those sources or leave them in.

Reviewer 2: ```suggestion
The exception is video and audio files. It has the option to remove the link to those sources or leave them in.
```
